### PR TITLE
master performance: 264 build new instances from a base image; improve multi user usage

### PR DESCRIPTION
**Pull request by daniwe4**
Doil now builds on the first create call a base image. This image is the template for following instances. This will minimize the build time for round about 3 minutes.

Debian is now pinned to 11 in all docker files.

**Multi User Support**

If you add a new doil user, doil also create a .config/composer folder. This is necessary for mounting it into the container.

Doil create checks for an .ssh folder in your home directory.

Doil forbids global instances below the home directory.

Doil adds or append a .gitconfig file for new users with safe directories for each global instance.

### CODE REVIEW

**Review by tbongers-cat — APPROVED** [review]
Looks good! As always, just some small notes.

File: app/src/Lib/Logger/LoggerFactory.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ class LoggerFactory
     public function getDoilLogger(string $channel) : LoggerInterface
     {
         $logger = new Logger($channel);
-        $stream = new StreamHandler(self::DOIL_LOG_PATH, Logger::DEBUG);
+        $stream = new StreamHandler(self::DOIL_LOG_PATH, Logger::DEBUG, true, 0777);
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Logger/LoggerFactory.php:20]: This is the only solution that works with multi users. I try it with 0770, but this doesn't work.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,7 @@ however **doil** needs [Docker](https://www.docker.com/) in order to work:
 * php*.*-dom
 * composer version = depending on installed php version
 * git
+* .ssh folder in your home directory. **doil** will mount it into the container.
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply README.md:54]: Done!

File: app/src/Commands/Instances/CreateCommand.php [hunk 3]
```diff
@@ -126,10 +144,15 @@ public function execute(InputInterface $input, OutputInterface $output) : int
...
-        // update debian image
-        $this->writer->beginBlock($output, "Updating debian image");
-        $this->docker->pull("debian");
-        $this->writer->endBlock();
+        // install base image if not exists
+        if ($this->docker->getImageIdsByName("doil/base_global")[0] == "") {
+            $this->writer->beginBlock($output, "Updating debian image");
+            $this->docker->pull("debian");
+            $this->writer->endBlock();
+            $this->writer->beginBlock($output, "Install base image");
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Commands/Instances/CreateCommand.php:152]: Done!

File: app/src/Lib/Docker/DockerShell.php [hunk 4]
```diff
@@ -350,7 +350,7 @@ public function pull(string $name) : void
         $cmd = [
             "docker",
             "pull",
-            $name . ":stable"
+            $name . ":11"
```
[the remark this answers is not in the document]
Comment (daniwe4) [reply app/src/Lib/Docker/DockerShell.php:353]: Done!

**Pull request merged**